commons/src/main/java/io/hyscale/commons/commands/CommandExecutor.java

```diff
@@ -122,101 +120,81 @@ public static boolean executeInDir(String command, File commandOutputFile, Strin
-        logger.debug("Executing command {} in dir {}", command, dir);
-        processBuilder.command(command.split(" "));
-        processBuilder.redirectErrorStream(true);
-        if (file != null) {
-            HyscaleFilesUtil.createEmptyFile(file);
-            processBuilder.redirectOutput(file);
-        }
-        Process process = processBuilder.start();
-        handleStandardInput(process, command, stdInput);
+        CommandLine commandToExecute = CommandLine.parse(command);
 
-        boolean readOutput = false;
-        StringWriter strWriter = new StringWriter();
-        if (file == null) {
-            readOutput = copyOutput(process, strWriter);
-        }
+        Executor executor = new DefaultExecutor();
+
+        // Output handler
+        OutputStream outputStream = getOutputStream(outputFile);
+
+        // Input handler
+        InputStream inputStream = getInputStream(stdInput);
+
+        // Stream handlers
+        PumpStreamHandler streamHandler = new PumpStreamHandler(outputStream, outputStream, inputStream);
+
+        // Timeout
+        ExecuteWatchdog watchDog = new ExecuteWatchdog(DEFAULT_TIMEOUT_IN_MILLIS);
+
+        executor.setStreamHandler(streamHandler);
+        executor.setWorkingDirectory(new File(dir));
+        executor.setWatchdog(watchDog);
+        int exitCode = 1;
         try {
-            exitCode = process.waitFor();
-        } catch (
-                InterruptedException e) {
-            logger.error("Error while waiting for process to complete");
-            throw e;
+            exitCode = executor.execute(commandToExecute);
+        } catch (ExecuteException e) {
+            // timeout case
+            exitCode = e.getExitValue();
         }
-        if (readOutput) {
-            cmdResult.setCommandOutput(strWriter.toString());
+        if (outputFile == null) {
+            cmdResult.setCommandOutput(getCommandOutput(outputStream));
         }
+
         cmdResult.setExitCode(exitCode);
         return cmdResult;
     }
 
-    /**
-     * Copy output from process input stream to string writer
-     *
-     * @param process
-     * @param strWriter
-     * @return is thread started
-     */
-    private static boolean copyOutput(Process process, StringWriter strWriter) {
-        return ThreadPoolUtil.getInstance().execute(() -> {
-            try {
-                do {
-                    IOUtils.copy(process.getInputStream(), strWriter, StandardCharsets.UTF_8);
-                } while (process.isAlive());
-            } catch (IOException e) {
-                logger.error("Error while reading command output", e);
-            }
-        });
+    private static String getCommandOutput(OutputStream outputStream) throws UnsupportedEncodingException {
+        if (outputStream != null && outputStream instanceof ByteArrayOutputStream) {
+            return ((ByteArrayOutputStream) outputStream).toString(ToolConstants.CHARACTER_ENCODING);
+        }
+        return null;
```

Comment:
Create a an outputstream impl which writes to string which uses a StringBuilder to append the writes. 
This applies only when user needs the result of the output